nfs/src/rpc_task.cpp

```diff
@@ -2237,6 +2315,13 @@ void rpc_task::free_rpc_task()
         bc_vec.clear();
         rpc_api->parent_task = nullptr;
         break;
+    case FUSE_WRITE:
+    case FUSE_FLUSH:
+        if (rpc_api) {
+            delete rpc_api;
+            this->rpc_api = nullptr;
+        }
```

Comment:
why do u want to delete task->rpc_api.
we don't delete this as it can be reused and we don't want to pay the allocation cost on every task create
Instead you can do cleanup if any in rpc_api->release() above.

Reply:
ASAN give error of memory leak.

Reply:
Ok, then it should not be delete from here but from the rpc_task_helper destructor.